src/vm/i386/asmhelpers.S

```diff
@@ -1205,14 +1205,196 @@ NESTED_ENTRY BackPatchWorkerAsmStub, _TEXT, NoHandler
+    push    eax                 //                                                  -- struct eax field
+    push    0                   // Create buffer space in the structure             -- struct floatingPointValuePresent field
+    push    0                   // Create buffer space in the structure             -- struct floatBuffer field
+    push    0                   // Create buffer space in the structure             -- struct doubleBuffer2 field
+    push    0                   // Create buffer space in the structure             -- struct doubleBuffer1 field
+    push    0                   // Create buffer space in the structure             -- struct functionId field
+
+    mov     edx, esp            // the address of the Platform structure
+    mov     ecx, [esp+52]       // The functionIDOrClientID parameter that was pushed to FunctionEnter
+                                // Skip past ProfilePlatformSpecificData we pushed: 40 bytes
+                                //   - plus saved edi, esi : 8 bytes   
+                                //   - plus return address from caller: 4 bytes   
+
+    #define STACK_ALIGN_PADDING 12
+    sub     esp, STACK_ALIGN_PADDING
+
+    CHECK_STACK_ALIGNMENT
+    call    C_FUNC(ProfileEnter)
+
+    add     esp, STACK_ALIGN_PADDING
+    #undef  STACK_ALIGN_PADDING
+
+    add     esp, 20             // Remove buffer space
+    pop     eax
+    pop     edx
+    pop     ecx
+    add     esp, 8              // Remove buffer space
+    pop     edi
+    pop     esi
     ret
 NESTED_END ProfileEnterNaked, _TEXT
 
 NESTED_ENTRY ProfileLeaveNaked, _TEXT, NoHandler
+    push    ecx                 // We do not strictly need to save ECX, however
+                                // emitNoGChelper(CORINFO_HELP_PROF_FCN_LEAVE) returns true in the JITcompiler
+    push    edx                 // Return value may be in EAX:EDX
+
+    //
+    // Push in reverse order the fields of ProfilePlatformSpecificData
+    //
+    push    dword ptr [esp+8]   // EIP of the managed code that we return to.	-- struct ip field
+    push    ebp                 // Methods are always EBP framed
+    add     dword ptr [esp], 8  // Skip past the return IP, straight to the stack args that were passed to our caller
+                                // Skip past saved EBP value: 4 bytes
+                                //   - plus return address from caller's caller: 4 bytes   
+                                //
+                                // Assuming Foo() calls Bar(), and Bar() calls ProfileEnterNake() as illustrated (stack 
+                                // grows up). We want to get what Foo() passed on the stack to Bar(), so we need to pass 
+                                // the return address from caller's caller which is Foo() in this example.
+                                //
+                                // ProfileEnterNaked()
```

Comment:
Shouldn't this be `ProfileLeaveNaked`? Also 4 lines above.

Reply:
Sure, should be ProfileLeaveNaked. Thanks!